### PR TITLE
Dist branch

### PR DESCRIPTION
**Pull request by aekong**
* create a distribution branch that contains the following 
  * compiled css and js
  * component source files
    * .hbs template files
    * .scss source files

Ideally, this will be the branch referenced by projects intending to use the design system, so that the scss files can be used for imports to extend the consuming projects styles

### CODE REVIEW

**Comment by aekong** [comment]
@jrubenoff this update basically creates/pushes to this distribution branch: https://github.com/SFDigitalServices/sf-design-system/tree/distribution and creates a tag to it that can be referenced when including the design system as a dependency

File: .circleci/scripts/deploy.sh [hunk 1]
```diff
@@ -31,11 +36,14 @@ git add -A
 terminus -n auth:login --machine-token="$TERMINUS_MACHINE_TOKEN"
 
 if [ $CIRCLE_BRANCH == $SOURCE_BRANCH ]; then
-  git commit -m "automated static site deploy: ${CIRCLE_SHA1}" --allow-empty
+  git commit -m "automated static site deploy: ${GIT_COMMIT_MSG}" --allow-empty
   git push origin -f $SOURCE_BRANCH:$TARGET_BRANCH # push to gh-pages
   git push -f pantheon $SOURCE_BRANCH # push to pantheon master
+  git checkout $SOURCE_BRANCH
```
Comment (josh-chou) [inline .circleci/scripts/deploy.sh]: is this redundant?  didn't we just push?

File: .circleci/scripts/deploy.sh [hunk 2]
```diff
@@ -53,4 +61,15 @@ else
...
\ No newline at end of file
+fi
+
+# copy src back in, remove unnecessary things, commit, tag, and push
+git checkout distribution || git checkout --orphan distribution
+cp -r ../src .
+rm -rf *.txt
+rm -rf components themes *.html
+git add -A
+git commit -m "distribution build: ${GIT_COMMIT_MSG}" --allow-empty
```
Comment (josh-chou) [inline .circleci/scripts/deploy.sh]: so a distribution build gets checked in and tagged with every build?  even when working on dev branches?

**Review by josh-chou — CHANGES_REQUESTED** [review]

File: .circleci/scripts/deploy.sh (hunk 1 above)
Comment (aekong) [reply .circleci/scripts/deploy.sh]: Yeah you’re right. Redundant. 😞 Thanks for catching.

File: .circleci/scripts/deploy.sh (hunk 2 above)
Comment (aekong) [reply .circleci/scripts/deploy.sh]: Yeah, think it’s overkill? My intent was to have a tag for package managers to reference, but maybe it should just be for master?

Comment (josh-chou) [reply .circleci/scripts/deploy.sh]: i think it should be just for master, but you are already tagging a version for master.
what the difference between the version in master and distribution build version?

Comment (aekong) [reply .circleci/scripts/deploy.sh]: master branch has everything (gulp build tasks, etc), distribution only has the dist css js and images and the source scss and template files

Comment (josh-chou) [reply .circleci/scripts/deploy.sh]: can they have the same version?
move the code to cleanup build artifacts into the master if clause and just have one version.
unless you foresee a need to have versions for development builds.

Comment (aekong) [reply .circleci/scripts/deploy.sh]: Yeah now that you mention it I think want to rework this thing.  Thanks for the feedback!

**Comment by aekong** [comment]
Ok, I _think_ this is better.

The deploy script should now (on push to master)
- tag `master` with the package.json version
- export the static site to `gh-pages` and pantheon master
- create a distribution branch and tag it with the package.json version + `-dist` (eg: `1.1.0-dist`)

If the tag (or variation of tag) already exists when pushing to `master`, the build will fail early

**Review by josh-chou — APPROVED** [review]

**Comment by aekong** [comment]
woo hoo, now to see if actually does what it should

**Pull request merged**